### ISSUE TITLE
Implement a remove_parameter_handler function

### ISSUE THREAD

**Issue by mattwthompson** [openforcefield/openforcefield#552]
**Is your feature request related to a problem? Please describe.**
We would like to be able to remove a `ParameterHandler` from a force field, i.e. the opposite of `ff.register_parameter_handler`

**Describe the solution you'd like**
In:
```python3
from openforcefield.typing.engines.smirnoff.forcefield import ForceField
from openforcefield.typing.engines.smirnoff.parameters import BondHandler

# Load up a bare force field
ff = ForceField()
print(len(ff._parameter_handlers))

# Add a parameter handler
bh = BondHandler(skip_version_check=True)
ff.register_parameter_handler(bh)

print(len(ff._parameter_handlers))

# Would prefer doing something like
# ff.remove_parameter_handler('Bonds')
del ff._parameter_handlers['Bonds']

print(len(ff._parameter_handlers))
```
Out
```
0
1
0
```
**Describe alternatives you've considered**
It's possible to do this by accessing `ff._parameter_handlers` directly but that is not desirable

**Additional context**
This is probably not something commonly encountered, so it may be a low priority. But this could help flexibility and modularity, i.e. swapping out a handler after loading up an XML.

I'm fairly new to this API so there's a chance I've completely missed an existing function

**Comment by mattwthompson** [comment]
I think https://github.com/openforcefield/openforcefield/pull/678 will fix this

**Comment by j-wags** [comment]
Closed by #678
